### PR TITLE
Make dayTimeMode optional

### PR DESCRIPTION
**Pull request by cquiroz**
This is to make observe compatible with the TCC

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/model/GuideConfig.scala [hunk 1]
```diff
@@ -59,7 +59,7 @@ object GuideConfig {
         MountGuideOption.MountGuideOff,
         M1GuideOff,
         M2GuideOff,
-        false,
+        None,
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/model/GuideConfig.scala:62]: How is a `None` value interpreted?

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/model/GuideConfig.scala:62]: false, but it is not really used on observe

**Review by swalker2m — APPROVED** [review]
Seems like it shouldn't be an option in Navigate / TCC ?  At any rate :+1:

**Comment by cquiroz** [comment]
In navigate as yyou said it shouldn't be an option but we don't wantn to modify TCC
I think it should return to `Option` once we kill TCC

**Pull request merged**